### DISCUSSION TITLE
Another RP2040 board that I can't get to work

### DISCUSSION THREAD

**Discussion by fryc88** [earlephilhower/arduino-pico#2271] [General]
Hi,
I have tried I think all possible rp2040 boards through your lib and none of them will get my board to work. This is a board from 3d printers world, FYSETC brand, board called ERB v1.
Based on RP2040 chip, with 2x TMC2209 built in, working in uart mode. I wanted to give it a go with small project.
Could you please advise what i need to look at when choosing boards? I would assume chip is chip and as long as pins are correct it should work, but it does not apparently :(
That's the board:
https://github.com/FYSETC/FYSETC-ERB/tree/main/V1.0/hardware

![IMG_2520](https://github.com/user-attachments/assets/68389b95-314b-4ea3-a8b5-75ecf6036af4)

**Comment by earlephilhower** [comment]
Moved to a discussion because it's more of a general question and not something w/the core.

Have you built and run code on a regular Pico board?  Under Windows there's some USB port configuration needed (Zadig, see the docs) so Windows will show a serial port.

What exactly doesn't work?  When you hold bootsel and plug in, do you get the drive for upload?  Does a simple `void setup() {}  void loop() { Serial.println("hello world"); delay (1000);} ` not even bring up a USB serial port?  

Have you tried the Generic board with Boot2 as `Generic SPI / 4`?  Also make sure you set the flash size correctly.  Too small is OK, too large is going to cause problems, potentially.

**Reply by fryc88** [reply]
Well, to be fair I've been playing with it later on and right now its not recognised via usb anymore. I was trying to figure out if gpios are mapped differently or what, and when I went to like 4-6 it just stopped showing via usb and current draw from PSU is smaller, about 0.02A now, so I either fried board or somehow messed up bootloader.

Thing that was not working - basically it did flash sample code from tmc2209 library, I've chosen pins and absolutely nothing was happening, not even blink led.
Then I tried Arduino nano rp2040 from different library and some bits worked but not serialpio or software serial, so couldn't get demanded code to work there.
